R/transformCounts.R

```diff
@@ -359,17 +361,22 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
 
 ###############################.apply_pseudocount###############################
 # This function applies pseudocount to abundance table.
-.apply_pseudocount <- function(mat, pseudocount){
+.apply_pseudocount <- function(mat, pseudocount, na.rm = TRUE, ...){
     if( .is_a_bool(pseudocount) ){
-        # If pseudocount TRUE but some NAs or negative values, numerical pseudocount needed
-        if ( pseudocount && (any(is.na(mat)) || any(mat < 0, na.rm = TRUE)) ){
-            stop("The assay contains missing or negative values. ",
+        # If pseudocount TRUE and some NAs, a warning is issued
+        if ( pseudocount && any(is.na(mat)) ){
+            warning("The assay contains missing values (NAs). These will be 
+                ignored in pseudocount calculation.", call. = FALSE)
+        }    
+        # If pseudocount TRUE but some negative values, numerical pseudocount needed
+        if ( pseudocount && any(mat < 0, na.rm = TRUE) ){
+            stop("The assay contains missing values (NAs) or negative values. ",
```

Comment:
This message is now incorrect. It should be just "The assay contains negative values.."

After this fix, good to go